### ISSUE TITLE
Ability to cycle between selectable objects (using a keyboard shortcut)

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5867]
Issue imported from https://tracker.freecad.org/view.php?id=3786

* **Reporter:** Moult
* **Date submitted:** 1/21/2019
* **FreeCAD version:** 0.18
* **Category:** Feature
* **Status:** new
* **Tags:** shortcuts

# Original report text

Sometimes, I am unable to select an object or edge. This is because that edge or object may be coincident with another edge or object. I can usually get around this by hiding objects in the tree, or by changing into wireframe mode, or both. This works, but is slow and not user friendly.

In Revit, you can press "tab" to cycle between objects to select under your mouse until you have the right object. In Blender, you can keep on clicking to select until you select the right one and then perform the operation.

I propose that a similar feature such as hitting tab be implemented. I suspect the Blender approach of clicking again will not work, because FreeCAD implementations select the operation beforehand, and then select the object. Therefore some keypress should occur. "Tab" seems sensible (after all, alt-tab is switch windows in window management)

Forum discussion is here:
https://forum.freecadweb.org/viewtopic.php?p=281380

# Other bug information

* **Priority:** normal
* **Severity:** feature
* **Category:** Feature
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by Moult 2019-02-07 20:44
PR for new feature to cycle between selectable objects: https://github.com/yorikvanhavre/BIM_Workbench/pull/19
PR for new feature to cycle between snappable objects that overlap: https://github.com/FreeCAD/FreeCAD/pull/1943

This bugreport can be closed :)

**Comment by vmario89** [comment]
waiting for that feature a long time. really important to undo stuff in a sketch :-/

**Comment by luzpaz** [comment]
@yorikvanhavre looks like this was implemented in https://github.com/yorikvanhavre/BIM_Workbench/pull/19  
But just in BIM workbench, right ?

**Comment by vmario89** [comment]
seems so. i did not find any key combination to cycle through overlayed elements in sketcher, which is crucial

**Comment by yorikvanhavre** [comment]
Yes it is only implemented in BIM. I haven't checked for a long time if it's til working, though.
In theory this should work with any kind of selection, even outside the BIM WB, but this needs to be tested. Also I'm not sure how far it can handle sketch-specific elements inside the sketcher's edit mode.
